js/ui/calendar/ui.calendar.js

```diff
@@ -835,9 +890,20 @@ const Calendar = Editor.inherit({
     },
 
     _navigatorClickHandler: function(e) {
-        const currentDate = this._getDateByOffset(e.direction, this.option('currentDate'));
+        const { currentDate, views } = this.option();
+        let offset = e.direction;
```

Comment:
Let't move offset calculation logic to separate function, something like this

_getNavigateOffset(offset) {
    const { views } = this.option();

    if(views > 1) {
            const additionalViewActive = this._isAdditionalViewDate(currentDate);
            const shouldDoubleOffset = additionalViewActive && offset < 0 || !additionalViewActive && offset > 0;

            if(shouldDoubleOffset) {
                return 2 * offset;
            }
        }

       return offset;
}